src/frontends/pytorch/src/op/scatter.cpp

```diff
@@ -17,58 +17,110 @@ namespace op {
 
 using namespace ov::op;
 
+namespace {
+Output<Node> prepare_source(const NodeContext& context, Output<Node> src, Output<Node> index, Output<Node> input) {
+    auto src_partial_shape = src.get_partial_shape();
+    auto index_shape_rank = get_shape_rank(context, index);
+    auto index_shape = std::get<0>(index_shape_rank);
+    auto index_rank = std::get<1>(index_shape_rank);
+
+    // Source input can be either Tensor which should be passed in original shape or Scalar that should be broadcasted
+    // into shape of indices.
+    // TODO: Figure out way to dynamically broadcast scalar src only, without affecting Tensor src. Current
+    // implementation will fail if Scalar source would have dynamic rank.
+    if (src_partial_shape.rank().is_static() && src_partial_shape.rank().get_length() == 0) {
+        src = context.mark_node(std::make_shared<v3::Broadcast>(src, index_shape));
+    }
+
+    auto const_0 = context.mark_node(v0::Constant::create(element::i32, Shape{}, {0}));
+    auto zeros = context.mark_node(std::make_shared<v3::Broadcast>(const_0, index_rank));
+    auto const_1 = context.mark_node(v0::Constant::create(element::i32, Shape{}, {1}));
+    auto ones = context.mark_node(std::make_shared<v3::Broadcast>(const_1, index_rank));
+    // In torch indices can be of different shape than source tensor. Create slice to trim source tensor to shape of
+    // indices.
+    auto src_pruned = context.mark_node(std::make_shared<v8::Slice>(src, zeros, index_shape, ones));
+
+    auto src_input_dtype = context.mark_node(std::make_shared<v1::ConvertLike>(src_pruned, input));
+    return src_input_dtype;
+};
+const std::unordered_map<std::string, v12::ScatterElementsUpdate::Reduction> TORCH_REDUCTION_TO_OV{
+    {"add", v12::ScatterElementsUpdate::Reduction::SUM},
+    {"multiply", v12::ScatterElementsUpdate::Reduction::PROD},
+    {"sum", v12::ScatterElementsUpdate::Reduction::SUM},
+    {"prod", v12::ScatterElementsUpdate::Reduction::PROD},
+    {"mean", v12::ScatterElementsUpdate::Reduction::MEAN},
+    {"amax", v12::ScatterElementsUpdate::Reduction::MAX},
+    {"amin", v12::ScatterElementsUpdate::Reduction::MIN}};
+
```

Comment:
better to use:
```
const A & get_a() {
   static const A a;
   return a;
}
```
to define an order of variables creation. So, `a` is created only when it's used

Reply:
applied, thanks

Reply:
Doesn't seem as fully applied... It doesn't return constref and it doesn't create static const inside

Reply:
now fully :)